neqo-transport/src/streams.rs

```diff
@@ -131,6 +131,8 @@ impl Streams {
                 stats.reset_stream += 1;
                 if let (_, Some(rs)) = self.obtain_stream(*stream_id)? {
                     rs.reset(*application_error_code, *final_size)?;
+                } else if !self.ensure_existed_if_local(*stream_id) {
+                    return Err(Error::StreamStateError);
                 }
```

Comment:
The negation and the implicit `false` on remote streams, plus the repeated `return` is hard for me to parse.

What do you think of https://github.com/larseggert/neqo/pull/33?

Reply:
Like it! Merged.

Reply:
Except it doesn't work :-(

Reply:
The `!self.ensure_existed_if_local` condition is only sufficient _iff_ `obtain_stream` didn't match.

Reply:
@mxinden see 19e7ed59d28341c50351a37534d33d7c01d8f7a8 for a different approach.

Reply:
Oh sorry for the trouble. New version looks good.